library/src/main/java/org/apache/apex/malhar/lib/window/impl/KeyedWindowedOperatorImpl.java

```diff
@@ -135,6 +149,15 @@
   }
 
   @Override
+  public void endWindow()
+  {
+    super.endWindow();
+    if (updatedKeyStorage != null && isFiringOnlyUpdatedPanes()) {
```

Comment:
I think it's better to have this as a method returning a boolean since it's used in a lot of places.